lib/rendition.js

```diff
@@ -60,6 +155,26 @@ class Rendition {
         return obj;
     }
 
+    inline() {
+        return Number.isInteger(this.instructions.inlineLimit)
+            && this.instructions.inlineLimit <= INLINE_LIMIT_MAX
+            && this.size() <= this.instructions.inlineLimit;
+    }
+
+    asDataUri() {
+        // should never happen as inlineLimit is limited, but extra safeguard against reading large files
+        if (this.size() > DATA_URI_LIMIT) {
+            throw new GenericError(`Rendition too large for data uri ${this.name}`);
+        }
+        const data = fs.readFileSync(this.path);
+
+        // cannot have spaces in data uri
+        // convert 'text/plain; charset=utf-8' into 'text/plain;charset=utf-8'
+        const type = this.contentType().replace(" ", "");
+
+        return `data:${type};base64,${data.toString("base64")}`;
```

Comment:
Do we want to add a safety layer verifying the data uri is valid before sending? E.g. using https://www.npmjs.com/package/valid-data-url.

Reply:
We are building it according to the spec here...